target/linux/ar71xx/base-files/lib/ar71xx.sh

```diff
@@ -582,7 +582,15 @@ ar71xx_board_detect() {
 		;;
 	*"CPE510/520")
 		name="cpe510"
-		tplink_pharos_board_detect "$(tplink_pharos_get_model_string | tr -d '\r')"
+		tplink_pharos_board_detect "$(tplink_pharos_v2_get_model_string)"
```

Comment:
Thanks for tidying up; much clearer now.
However, I do not like the multiple occasions of tplink_pharos_board_detect. I would prefer to do the evaluation based on the model string, so save "$(tplink_pharos_v2_get_model_string)" into a variable:
modelstr="$(tplink_pharos_v2_get_model_string)"
and then do some case $modelstr in 'something' instead of your case $AR71XX_MODEL.
I don't thick its a good idea to use non-local variables for this local switch-case.
The same would be applicable for platform.sh.
However, I'm not a committer, so others may see it differently.